PanelViewController/PanelViewController.swift

```diff
@@ -125,6 +154,23 @@ class PanelViewController: UIViewController {
         return panelViewHeight
     }
     
+    fileprivate func performStateChange(velocity: CGPoint) {
+        if showsMidState {
```

Comment:
What about passing the velocity to the togglePaneState, and having a three pronged switch there?  So this func would just be:
`togglePaneState(velocity: velocity)`
 `animatePane(velocity: velocity)`
then have a switch statement in togglePaneState that includes the updatePaneState() for the midstate condition